test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/Expressions/FilterBinderTests.cs

```diff
@@ -2956,6 +2956,25 @@ public void FilterByDynamicProperty()
                 "$it => (Convert(IIF((($it.ProductProperties != null) AndAlso $it.ProductProperties.ContainsKey(Token)), $it.ProductPropertiesToken, null)) == \"1\")");
         }
 
+        [Fact]
+        public void InOnPrimitiveCollectionPropertyOnRHS()
+        {
+            var filters = VerifyQueryDeserialization(
+               "42 in AlternateIDs",
+               "$it => $it.AlternateIDs.Contains(42)",
+               NotTesting);
+
+            RunFilters(
```

Comment:
What exactly does `RunFilters` do here?

Reply:
`RunFilters` is a shared method that we use to do some common assertions - instead of repeating the logic all over. If you look at the first `RunFilters` call, the array `new[] { 1, 2, 42 }` has the target value of `42` so we'll verify that the outcome of the filter expression will be `true` with and without null propagation. In the second `RunFilters` call, the array doesn't have the target value so we'll verify that the outcome of the filter expression will be `false` with and without null propagation. [Here](https://github.com/OData/WebApi/blob/40b8e57fe9f29490cbbe04aadfb2acf3e4fd7638/test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/Expressions/FilterBinderTests.cs#L3014) is the method definition.

Reply:
What does `[With|Without]nullProgagation` mean/do?

Reply:
The null propagation operator in C# - `?`. It enables the us to check for the null value in an object reference chain - e.g. `product?.Name`. In this context, we'd generate two filter expressions - one [with null propagation and another without](https://github.com/OData/WebApi/blob/40b8e57fe9f29490cbbe04aadfb2acf3e4fd7638/test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/Expressions/FilterBinderTests.cs#L3087).
The result could be different for the two scenarios. So we test and verify the result against both scenarios.